doc/rst/language/spec/conversions.rst

```diff
@@ -191,48 +179,142 @@ subtype of the target type.
+
+      writeln(x);
+
+   .. BLOCK-test-chapeloutput
+
+      true
+
+   *Example (subtype-pass-int8-integral.chpl)*
+
+   Since ``int(8)`` is a subtype of ``integral``, the type ``int(8)`` can
+   be passed to the type argument ``type t: integral``
+   (:ref:`Legal_Argument_Mapping`). As a result the following program
+   will compile:
+
+   .. code-block:: chapel
+
+      proc f(type t: integral) { }
       f(int(8));
 
+   *Example (subtype-parent-class.chpl)*
+
+   This example demonstrates that ``ChildClass`` is a subtype of
+   ``ParentClass``.
+
+   .. code-block:: chapel
+
+     class ParentClass { }
+     class ChildClass : ParentClass { }
+
+     writeln(isSubtype(ChildClass, ParentClass)); // outputs true
+     writeln(isSubtype(borrowed ChildClass, borrowed ParentClass)); // outputs true
+
+     proc f(type t: ParentClass) { }
+     f(ChildClass); // implicit subtype conversion
+
+     proc g(type t: borrowed ParentClass) { }
+     g(borrowed ChildClass); // implicit subtype conversion
+
+     // The implicit subtype conversion can also apply to non-type arguments:
+     proc h(in arg: owned ParentClass) { }
+     h(new owned ChildClass()); // implicit subtype conversion
+
    .. BLOCK-test-chapeloutput
 
-      type-argument-conversion-error.chpl:2: error: unresolved call 'f(type int(8))'
-      type-argument-conversion-error.chpl:1: note: this candidate did not match: f(type t: int)
-      type-argument-conversion-error.chpl:2: note: because actual argument #1 with type 'int(8)'
-      type-argument-conversion-error.chpl:1: note: is passed to formal 't: int(64)'
+     true
+     true
 
-   *Example (type-argument-conversion.chpl)*
+   *Example (subtype-nilable.chpl)*.
 
-   In contrast, this code demonstrates an implicit conversion that
-   does succeed because a child class is a subtype of a parent class, and
-   an ``owned`` class type is a subtype of an undecorated (generic
-   management) class type.
+   This example shows that a non-nilable class type is a subtype of a
+   nilable class type with the same management.
+
+   .. code-block:: chapel
+
+      class C { }
+
+      writeln(isSubtype(C, C?)); // outputs true
+      writeln(isSubtype(owned C, owned C?)); // outputs true
+
+   .. BLOCK-test-chapeloutput
+
+     true
+     true
+
+   *Example (subtype-three.chpl)*.
+
+   This example demonstrates a combination of all three rules. Note that
+   ``ParentClass`` is a generic type meaning the class type
+   ``ParentClass`` with any memory management strategy
+   (:ref:`Class_Types`).
 
    .. code-block:: chapel
 
      class ParentClass { }
      class ChildClass : ParentClass { }
 
-     proc g(type t: ParentClass) { }
-     g(owned ChildClass);
+     writeln(isSubtype(ChildClass, ParentClass?)); // outputs true
+
+     proc f(type t: ParentClass?) { }
+     f(ChildClass); // uses implicit subtype conversion
+
+     proc g(in arg: ParentClass?) { }
+     g(new owned ChildClass()); // uses implicit subtype conversion
```

Comment:
We can transfer ownership from a variable that is otherwise dead (using the copy elision feature) so 

``` chapel
{
  var x = new owned ChildClass();
  g(x);
}
```

should work just fine (but it does require x not be global IIRC).